New package: civetweb-1.15

bingulo opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->

#### Local build testing
- I built this PR locally for my native architecture, (x86_664-LIBC)

classabbyamp commented on srcpkgs/civetweb/template:
```diff
@@ -0,0 +1,42 @@
+build_style=gnu-makefile
+make_install_target="install install-headers install-lib install-slib"
+make_use_env=yes
+hostmakedepends="openssl-devel"
```
```suggestion
```
is only used for headers/linking, not necessary on host

classabbyamp commented on srcpkgs/civetweb/template:
```diff
@@ -0,0 +1,42 @@
+			vmove usr/include
+				vmove "usr/lib/*.a"
+				vmove "usr/lib/*.so"
+				vmove "usr/lib/*.so.1"
```
```suggestion
```
this should be in the main package

classabbyamp commented on srcpkgs/civetweb/template:
```diff
@@ -0,0 +1,42 @@
+}
+
+civetweb-devel_package() {
+	depends="civetweb-${version}_${revision}"
```
```suggestion
	depends="${sourcepkg}>=${version}_${revision}"
```

classabbyamp commented on srcpkgs/civetweb/template:
```diff
@@ -0,0 +1,42 @@
+		}
+}
+
+civetweb-doc_package() {
```
I don't think there's enough documentation here to warrant a subpkg

```
/usr/share/doc/civetweb/CREDITS.md
/usr/share/doc/civetweb/LICENSE.md
/usr/share/doc/civetweb/README.md
/usr/share/doc/civetweb/RELEASE_NOTES.md
/usr/share/doc/civetweb/SECURITY.md
/usr/share/doc/civetweb/civetweb_64x64.png
/usr/share/doc/civetweb/index.html
```

bingulo replied:
I'll remove the doc subpkg, thx.

classabbyamp reviewed (commented):
there's some extra indentation in your subpackage definitions too

classabbyamp commented on srcpkgs/civetweb/template:
```diff
@@ -0,0 +1,32 @@
+
+civetweb-devel_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+		short_desc+=' - development files'
+		pkg_install() {
+			vmove usr/include
+			vmove "usr/lib/*.a"
+			vmove "usr/lib/*.so"
+		}
```
```suggestion
	short_desc+=' - development files'
	pkg_install() {
		vmove usr/include
		vmove "usr/lib/*.a"
		vmove "usr/lib/*.so"
	}
```

classabbyamp commented:
you'll need to squash this into 1 commit and force-push

bingulo commented:
Did I need to add libcivetweb.so.1 to common/shlibs?

classabbyamp commented:
yes, and I think there should be a `libcivetweb` subpackage too

```sh
libcivetweb_package() {
	short_desc+=" - library"
	pkg_install() {
		vmove usr/lib/*.so.*
	}
}
```

bingulo commented:
> yes, and I think there should be a `libcivetweb` subpackage too
> 
> ```shell
> libcivetweb_package() {
> 	short_desc+=" - library"
> 	pkg_install() {
> 		vmove usr/lib/*.so.*
> 	}
> }
> ```

Should I add this subpackage as dependency for main package? I tested and didn't find any issue without it.

classabbyamp commented:
no, it should be automatically detected

classabbyamp commented:
the devel package should be called libcivetweb-devel now

classabbyamp commented:
looks decent now, can you squash your commits and force-push?

bingulo commented:
> looks decent now, can you squash your commits and force-push?

I need to squash all commits from "New package" to now into one?

classabbyamp commented:
yes, it should all be in a single commit called `New package: civetweb-1.15`

classabbyamp commented:
hm I think something got messed up there

classabbyamp commented:
there's still a `libcivetweb-devel/var/db/xbps/pkgdb-0.38.plist` in there

bingulo commented:
> hm I think something got messed up there

My bad, I ran `git add srcpkgs` and added garbage.

I fixed depency of libcivetweb on libcivetweb-devel, where was wrong (was civetweb).

bingulo commented:
Now everything should be ok.

paper42 commented:
please remove the commit message body

bingulo commented:
> please remove the commit message body

done

bingulo commented:
I'll close this PR cause It's a dependency of another package I built, and it'll be referenced there #38537
